sm_cm_config/src/xml_generate.py

```diff
@@ -0,0 +1,255 @@
+#             remain = (start + i) % 2
+#             node.set('address', str(hex(addr)))
+#             i += 1 # offset for next sensor
+#             # this appears to be on all the nodes
+#             node.set("permission", "r")
+#             # set parameter based on the type
+#             if ( c['type'] == 'int8' ):
+#                 width=8
+#                 format = "d"
+#             elif ( c['type'] == 'fp16' ):
+#                 width=16
+#                 format = "fp16"
+#             elif ( c['type'] == 'char' ):
+#                 width=8
+#                 format = "c"
+#             elif ( c['type'] == 'uint32_t' ):
+#                 width=32
+#                 format = "u"
+#             else:
+#                 print("ERROR: unknown type", c['type'])
+#                 break
+#             mask = (1 << width) - 1
+#             if (remain == 0):
+#                 node.set('mask', "0x{0:08X}".format(mask))
+#             else:
+#                 node.set('mask', "0x{0:08X}".format(mask<<16))
+#             if 'extra' in c:
+#                 extra = c['extra']
+#             else:
+#                 extra = ""
+#             node.set('parameters', "format=" + format + ";" + extra)
+#         if 'size' in c:
+#             print("size:", c['size'], c['name'])
+#             node.set('size', str(hex(c['size'])))
+# 
+
+# %% [markdown]
+# 
+
+# %%
+tree = ET.ElementTree(cm)
+ET.indent(tree, space='\t')
+tree.write("test2.xml")
+
+# %%
+def calcSize(c: dict) -> int:
+    if 'size' in c:
+        size = c['size']*2 # extra factor of 2 for 16 to 32 bit
+    else:
+        size = 1
+    if '32' in c['type']:
+        size = size * 2
+    elif 'char' in c['type']:
+        size = size // 2
+    return size
+
+
+# %%
+# check overlaps
+# create an object with a name, and a start end end register
+# do so for every entry
+# then ensure that no two objects overlap
+
+# create a class to hold a name, start, end, and size
+# add a method to check if two objects overlap
+# and a pretty print method
+from itertools import chain
+class reg:
+    def __init__(self, name, start, end, size):
+        self.name = name
+        self.start = start
+        self.end = end
+        self.size = size
+    def __str__(self):
+        return "name: " + self.name + " start: " + str(self.start) + " end: " + str(self.end) + " size: " + str(self.size)
+    def __repr__(self):
+        return "name: " + self.name + " start: " + str(self.start) + " end: " + str(self.end) + " size: " + str(self.size)
+    def overlaps(self, other):
+        if (self.start <= other.start and self.end >= other.start):
+            return True
+        if (self.start <= other.end and self.end >= other.end):
+            return True
+        if (self.start >= other.start and self.end <= other.end):
+            return True
+        return False
+    def overloads(self, other):
+        if (self.start >= 237):
+            return True
+        return False
```

Comment:
"others" isn't used and so I am not sure that the test on line 254 does what you mean it to do?

Reply:
it was working when I tested it but you're right that this syntax doesn't look good. I'll fix it to :
```
def overloads(self):
    if (self.start >= 237):
        return True
    return False
```

and then just call with `overloads(entries[len(entries)-1])`

Reply:
it was a mistake from copying and pasting...

Reply:
ok, I still don't think we should have that number 237 in there, shouldn't it be something like the start + size of the entry compared to the max?

Reply:
This 237 was a pure experimental value. I tried a few different starting addresses (even though start + size < 256) of the last node but ran into a wrap-around issue until I found this value. So, we can think of this number as a safe maximum starting value for now?

Reply:
I'm not sure I understand what this is supposed to do and why if start + size < 256 it would create a problem? I'm a bit uncomfortable with this approach because it sounds like we don't actually understand what is going on.

Reply:
I honestly did not understand why the wrap-around could have caused when the last address is still within 256. I can demonstrate the issue again. However, since we will have more address space, I believe that we will come back to this script for future development.

Reply:
when xml file has a start node < 237 (shows an expected table)
```
CM_FFARGV_MON| IS_FF25GBS| IS_PRESENT|
               --|-----------|-----------|
       F1_FFL12CH|          0|         63|
        F1_FFLDAQ|         15|         15|
       F2_FFL12CH|          0|         63|
        F2_FFLDAQ|         15|         15|
```
when xml file has start node > 237 (shows unexpected table)
```
CM_FFARGV_MON| IS_FF25GBS| IS_PRESENT|
               --|-----------|-----------|
       F1_FFL12CH|         15|         15|
        F1_FFLDAQ|          0|         63|
       F2_FFL12CH|         15|         15|
        F2_FFLDAQ|         63|         15|
```
For the starting node, I meant this [line](https://github.com/apollo-lhc/sm_cm_config/blob/main/data/zynqmon_2.yml#L217)